imagemodal/mixins/fragment.py

```diff
@@ -2,14 +2,53 @@
 Mixin fragment/html behavior into XBlocks
 """
 from __future__ import absolute_import
+
 from django.template.context import Context
+from xblock.core import XBlock
 from xblock.fragment import Fragment
 
 
 class XBlockFragmentBuilderMixin(object):
     """
     Create a default XBlock fragment builder
     """
+    static_css = [
+        'view.css',
+    ]
+    static_js = [
+        'view.js',
+    ]
+    static_js_init = None
+    template = 'view.html'
+
+    def provide_context(self, context):
+        """
+        Build a context dictionary to render the student view
+
+        This should generally be overriden by child classes.
+        """
+        context = context or {}
+        context = dict(context)
+        return context
+
+    @XBlock.supports('multi_device')
+    def student_view(self, context=None):
```

Comment:
Provide a default student view that adds normal css/js files into the default template file.
You can override/extend paths.
This moves more logic out of each XBlock view and into the mixin.
Eventually, the mixin will be split out into a shared repo.